### ISSUE TITLE
"No associated endpoint" (POP-1105) reported for service

### ISSUE THREAD

**Issue by marians** [derailed/popeye#289]
**Describe the bug**

First of all, I'm not sure this is a bug or not. Maybe I'm getting something wrong.

Anyway, we have an Ingress, Service,  and pods connected to each other and working fine. But popeye reports an error `No associated endpoint` on the service resource.

To me this sounds like the service is not connected to any Pod. But in fact, it is working fine, and we are serving production requests via Ingress.

**Expected behavior**

I don't expect to see a bug here.

**Screenshots**

This is a snippet of the report printed for `popeye -n fe-docs`:

![image](https://github.com/derailed/popeye/assets/273727/f726013d-5d12-4e9e-a611-fb59d444bf52)

For comparison, here is some data retrieved in other ways:

```nohighlight
$ kubectl -n fe-docs get svc
NAME      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
fe-docs   ClusterIP   172.31.39.151   <none>        8080/TCP   10d
```

```yaml
# kubectl -n fe-docs get svc -o yaml
apiVersion: v1
items:
- apiVersion: v1
  kind: Service
  metadata:
    annotations:
      meta.helm.sh/release-name: fe-docs
      meta.helm.sh/release-namespace: fe-docs
    creationTimestamp: "2024-02-20T11:51:06Z"
    labels:
      app: fe-docs
      app.kubernetes.io/managed-by: Helm
    name: fe-docs
    namespace: fe-docs
    resourceVersion: "48091506"
    uid: c171e1e2-2d63-4754-9f59-0aa1cd281503
  spec:
    clusterIP: 172.31.39.151
    clusterIPs:
    - 172.31.39.151
    internalTrafficPolicy: Cluster
    ipFamilies:
    - IPv4
    ipFamilyPolicy: SingleStack
    ports:
    - port: 8080
      protocol: TCP
      targetPort: http
    selector:
      app: fe-docs
    sessionAffinity: None
    type: ClusterIP
  status:
    loadBalancer: {}
kind: List
metadata:
  resourceVersion: ""
```

**Versions (please complete the following information):**
 - OS: mac OS 14.2.1
 - Popeye v0.20.3
 - K8s v1.25

**Comment by derailed** [comment]
@marians Thanks for the report Marian! 
1105 occurs if Popeye can't find associated endpoints. Which could be a bug since you've reported connectivity is fine??
What does `k get ep -n fe-docs` and `k get pods -l app=fe-docs -o wide` yields?

**Comment by marians** [comment]
```nohighlights
$ k get ep -n fe-docs
NAME      ENDPOINTS           AGE
fe-docs   100.64.19.78:8080   11d
```

```nohighlights
$ k get pods -n fe-docs -l app=fe-docs -o wide
NAME                       READY   STATUS    RESTARTS   AGE   IP             NODE                                            NOMINATED NODE   READINESS GATES
fe-docs-59d646c58c-mmfh4   1/1     Running   0          11d   100.64.19.78   ip-10-0-103-217.eu-central-1.compute.internal   <none>           <none>
```

**Comment by derailed** [comment]
@marians Thank you for the details!
Can't seem to repro on v0.20.4??
Are there any errors in Popeye's logs?  or any rbac rules to would prevent your user to list out endpoints?

**Comment by marians** [comment]
This doesn't occur in v0.20.4. ✨  Thanks

**Comment by marians** [comment]
Actually, it does occur in v0.20.4, too, but not on every execution.

**Comment by derailed** [comment]
Fixed v0.20.5
